### PR TITLE
Golint corrections for pkg

### PR DESCRIPTION
**Pull request by Jaanki**
golint corrections for all files in pkg folder

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

**Comment by mangelajo** [comment]
After applying your patch I'm unable to compile, I guess you have changed something on the NewK8sDatastore that conflicts main.go, may be you also need to lint-fix main.go?

# command-line-arguments
./main.go:104:7: cannot use "github.com/rancher/submariner/pkg/datastore/kubernetes".NewK8sDatastore(ss.ClusterID, stopCh) (type *"github.com/rancher/submariner/pkg/datastore/kubernetes".k8s) as type datastore.Datastore in assignment:
	*"github.com/rancher/submariner/pkg/datastore/kubernetes".k8s does not implement datastore.Datastore (wrong type for WatchClusters method)
		have WatchClusters(context.Context, []string, []string, func("github.com/rancher/submariner/pkg/types".SubmarinerCluster, bool) error) error
		want WatchClusters(context.Context, string, []string, func("github.com/rancher/submariner/pkg/types".SubmarinerCluster, bool) error) error
FATA[0032] exit status 2
make: *** [ci] Error 1

File: pkg/cableengine/ipsec/ipsec.go [hunk 1]
```diff
@@ -371,19 +371,19 @@ func (i *Engine) removeCableInternal(cableID string, client *goStrongswanVici.Cl
...
-	})
-	if err != nil {
-		klog.Errorf("Error when terminating child connection %s : %v", cableID, err)
-	}
+		err = client.Terminate(&goStrongswanVici.TerminateRequest{
+			Child: "submariner-child-" + cableID,
+		})
+		if err != nil {
+			klog.Errorf("Error when terminating child connection %s : %v", cableID, err)
+		}
```
Comment (tpantelis) [inline pkg/cableengine/ipsec/ipsec.go]: It looks like this commented code was further indented - was this intended?

File: pkg/cableengine/ipsec/ipsec.go [hunk 2]
```diff
@@ -371,19 +371,19 @@ func (i *Engine) removeCableInternal(cableID string, client *goStrongswanVici.Cl
...
-	})
-	if err != nil {
-		klog.Errorf("Error when terminating ike connection %s : %v", cableID, err)
-	} */
+		err = client.Terminate(&goStrongswanVici.TerminateRequest{
+			Ike: cableID,
+		})
+		if err != nil {
+			klog.Errorf("Error when terminating ike connection %s : %v", cableID, err)
+		} */
```
Comment (tpantelis) [inline pkg/cableengine/ipsec/ipsec.go]: same here

Comment (Jaanki) [reply pkg/cableengine/ipsec/ipsec.go]: Unindented.

File: pkg/cableengine/ipsec/ipsec.go (hunk 1 above)
Comment (Jaanki) [reply pkg/cableengine/ipsec/ipsec.go]: Unindented.

**Comment by Jaanki** [comment]
> After applying your patch I'm unable to compile, I guess you have changed something on the NewK8sDatastore that conflicts main.go, may be you also need to lint-fix main.go?
> # command-line-arguments
> 
> ./main.go:104:7: cannot use "github.com/rancher/submariner/pkg/datastore/kubernetes".NewK8sDatastore(ss.ClusterID, stopCh) (type *"github.com/rancher/submariner/pkg/datastore/kubernetes".k8s) as type datastore.Datastore in assignment:
> *"github.com/rancher/submariner/pkg/datastore/kubernetes".k8s does not implement datastore.Datastore (wrong type for WatchClusters method)
> have WatchClusters(context.Context, []string, []string, func("github.com/rancher/submariner/pkg/types".SubmarinerCluster, bool) error) error
> want WatchClusters(context.Context, string, []string, func("github.com/rancher/submariner/pkg/types".SubmarinerCluster, bool) error) error
> FATA[0032] exit status 2
> make: *** [ci] Error 1

It should compile now. Made few changes.

**Comment by Jaanki** [comment]
Images jankihc/submariner and jankihc/submariner-route-agent available at dockerhub.

**Pull request merged**